pyshader/py.py

```diff
@@ -165,8 +165,15 @@ def emit(self, opcode, *args):
 
         if opcode == "co_branch":
             assert not self._opcodes[-1][0].startswith("co_branch")
+
         self._opcodes.append((opcode, *args))
 
+        if opcode == "co_src_linenr":
+            if self._opcodes[-2][0] == "co_label":
+                # Swap the last two items so that a co_src_linenr does not
+                # get in between a co_label and a co_branch
+                self._opcodes.insert(-1, self._opcodes.pop(-1))
```

Comment:
This bug was detected during work on #53, and I extracted the fix, even though it contains concepts introduced in that PR :)

(The added code here takes care of another issue, albeit that issue causes the OR-collapsing to skip, which is how this whole problem was discovered.)